cmd/crank/beta/render/load.go

```diff
@@ -53,7 +53,15 @@ func LoadComposition(fs afero.Fs, file string) (*apiextensionsv1.Composition, er
 		return nil, errors.Wrap(err, "cannot read composition file")
 	}
 	comp := &apiextensionsv1.Composition{}
-	return comp, errors.Wrap(yaml.Unmarshal(y, comp), "cannot unmarshal composition resource YAML")
+	if err := yaml.Unmarshal(y, comp); err != nil {
+		return nil, errors.Wrap(err, "cannot unmarshal composition resource YAML")
+	}
+	switch gvk := comp.GroupVersionKind(); gvk {
+	case apiextensionsv1.CompositionGroupVersionKind:
+		return comp, nil
+	default:
+		return nil, errors.Errorf("not a composition: %s/%s", gvk.Kind, comp.GetName())
```

Comment:
```suggestion
		return nil, errors.Errorf("expecting a \"Composition\" as the 2nd argument, but found: %s/%s", gvk.Kind, comp.GetName())
```

This could make the message more useful?

Reply:
The caller is already wrapping the error adding some more information (`cannot load functions from <filename>`) and we print the usage text in case of errors too, so I think there is enough context to spot the issue, I would rather avoid leaking the argument position down to this level as it feels brittle. WDYT?

Reply:
Yeah, that sounds good then.